src/shipping/components/ShippingZoneRatesPage/ShippingZoneRatesPage.tsx

```diff
@@ -129,7 +130,7 @@ export const ShippingZoneRatesPage: React.FC<ShippingZoneRatesPageProps> = ({
     triggerChange,
   } = useForm(initialForm, undefined, { confirmLeave: true, formId });
 
-  const [taxClassDisplayName, setTaxClassDisplayName] = React.useState(
+  const [taxClassDisplayName, setTaxClassDisplayName] = useStateFromProps(
```

Comment:
useStateFromProps is a hook that updates state every time initial value changes (most likely a prop). I think it should be deprecated because it uses deep comparisons, whereas in 95% of cases we just need to compare references. Let me know what you think, should I make a reference comparison alternative in this PR?

Reply:
First of all, I think that every function should explicitly describe what it takes on the input and what it returns on the output. Thus here, one could infer that if it contains `useState...()` in its name, it should work as `useState()`, while it doesn't. So, in such a case, yes, I would even agree if we prohibit using it until we elaborate other way of fulfilling its purpose, for now marking it as `@deprecated` in TSdoc comment over the function.

I believe the purpose of it mostly is to update a state when new data arrives from the API (mostly after a successful update) on the whole page. But when new data arrives, I think we should explicitly change state e.g. via `setState` or use other not implicit solution, so everyone knows clearly what happens and when.